pkg/operator/discover/discover.go

```diff
@@ -212,10 +274,46 @@ func ListDevices(context *clusterd.Context, namespace, nodeName string) (map[str
 			logger.Warningf("failed to unmarshal %s", deviceJson)
 			continue
 		}
-		devices[node] = d
+		for i := range d {
+			devices = append(devices, d[i])
+		}
 	}
-	logger.Debugf("devices %+v", devices)
+	logger.Debugf("devices in use %+v", devices)
 	return devices, nil
+
+}
+
+func FreeDevices(context *clusterd.Context, nodeName, clusterName string) error {
```

Comment:
this removes the CM for a node? Either a comment or more descriptive method name would help

Reply:
added comments